src/lib.rs

```diff
@@ -1,17 +1,27 @@
 #[macro_use] extern crate bitflags;
 #[macro_use] extern crate lazy_static;
 
+extern crate hmac;
 extern crate num_bigint;
 extern crate num_integer;
 extern crate num_traits;
-extern crate ring;
+extern crate pbkdf2;
+extern crate sha2;
 
 use num_bigint::BigUint;
 use num_integer::Integer;
 use num_traits::cast::ToPrimitive;
 
-use ring::{digest, digest::Algorithm, hmac, pbkdf2};
+use hmac::{Mac, Hmac};
+use pbkdf2::pbkdf2;
+use sha2::{Sha256, Sha384, Sha512};
 
+#[derive(PartialEq, Eq)]
```

Comment:
Please add a (short doc comment) above. It doesn't need to be very descriptive, but since it's in the public API I'd like to have a doc for it.